### ISSUE TITLE
false positive: E1111 assignment-from-no-return when using function name "lazy"

### ISSUE THREAD

**Issue by gibiansky** [PyCQA/pylint#3586]
### Steps to reproduce
Run `pylint` on the following code:

```python
def lazy(func):                                          
    def wrapped(*args, **kwargs):                        
        return None                                      
                                                         
    return wrapped                                       
                                                        
@lazy                                                    
def function(y):                                         
    return y                                             
                                                  
has_an_error = function("this line has E1111")           
                                             
def lazy2(func):                                         
    def wrapped(*args, **kwargs):                        
        return None                                      
                                                         
    return wrapped                                       
                                             
@lazy2                                                   
def function2(y):                                        
    return y                       
                  
has_no_error = function2("this line does not have E1111")
```

### Current behavior
Produces E1111: assignment-from-no-return.

If you rename "lazy" to any other string, it goes away.

### Expected behavior
No false positive.

### pylint --version output
```
pylint 2.5.0
astroid 2.4.0
Python 3.8.2 (default, Apr 30 2020, 23:29:39)
[GCC 7.5.0]
```

**Comment by gibiansky** [comment]
It's being caused by this code in astroid, I think:

```
# List of possible property names. We use this list in order     
# to see if a method is a property or not. This should be        
# pretty reliable and fast, the alternative being to check each  
# decorator to see if its a real property-like descriptor, which 
# can be too complicated.                                        
# Also, these aren't qualified, because each project can         
# define them, we shouldn't expect to know every possible        
# property-like decorator!                                       
POSSIBLE_PROPERTIES = {                                          
    "cached_property",                                           
    "cachedproperty",                                            
    "lazyproperty",                                              
    "lazy_property",                                             
    "reify",                                                     
    "lazyattribute",                                             
    "lazy_attribute",                                            
    "LazyProperty",                                              
    "lazy",                                                      
    "cache_readonly",                                            
}                                                                
```

I've confirmed that this also happens if I use the name `reify` or `cachedproperty`.

**Comment by PCManticore** [comment]
Thanks for the report, this is definitely something we have to change. The hardcoded list of "possible" properties is sometimes wrong and can result in inferring the wrong thing depending on what names you use.
